RFCs/RFC_2.md

```diff
@@ -0,0 +1,48 @@
+# RFC: Branch and PR naming and tagging convention
+
+**Primary author(s)**: Giovan Angela
+
+**Created**: 10-10-2022
+
+**Last updated**: 10-10-2022
+
+**Approver(s)**:
+
+| Name | yes/not yet | date |
+| ---- | ----------- | ---- |
+|      | [ ]         |      |
+
+## Overview
+<!-- If someone only reads this far, what do you want them to know? -->
+Use one convention when naming branches or PR's.
+
+## Goal
+<!-- What problems are you trying to solve? What problems are you not trying to solve? -->
+The goal is to have one naming convention for PR's and branches such that it looks nice in git(hub).
+
+## Motivation
+<!-- What is the current state of the world? Why is this change being proposed? -->
+There are several types of names beings used without a clear structure. Some even use the name to indicate if it is an enchancement, feature, bugfix, etc... This does not look nice or organised.
+
+## Design
+<!-- How will this be implemented? -->
+Use the Jira board issue title as PR and branch title. For example this fictive issue for Summa (SUM-415) with description "Add documentation for undocumented code" will have a branch named "sum-415-add-documentation-for-undocumented-code" and it's PR will should be named "SUM-415 Add documentation for undocumented code". For branches and PRs that do not have an issue in Jira -> create a ticket in Jira.
+
+Set of labels that can be used to give an indication of the type of the PR:
+- Bugfix -> Something isn’t working
+- Discussion -> There is an open discussion which needs to be resolved first
+- Documentation -> An improvement/addition to the documentation
+- Enhancement -> An improvement to existing functionality OR the developer experience
+- Feature -> A new feature
+- Help Wanted -> The assignee encountered an issue and need help
+- Housekeeping -> Something that keeps or makes the codebase clean(er) 
+- Refactor -> A change to the codebase with no impact on the current behaviour
```

Comment:
How do these labels relate to the branch name that you're trying to standardize in this RFC?

Reply:
Currently working on a pre-commit hook (https://github.com/typicode/husky) which will run a check to see if 'current' branch is allowed. Let's agree, as Varias, on a list of allowed prefixes and a format of branch-name

Reply:
Currently have added bash-script into PR for intermin:
https://github.com/varias-io/intermin/pull/254

Reply:
> How do these labels relate to the branch name that you're trying to standardize in this RFC?

I, for one, am not fond of the usage of labels within the branch name, `"refactor/SUM-XYZ-something-foo-bar"`, due to the extra step required when using the autocomplete feature of my terminal. But, in order to still be able to differentiate between the types of branches, I'd like to use the PR labels as an indication of what a branch is supposedly doing. Another reason why I think the PR is the place for such labels and not the branch name is because a code review, in the ideal world, is done in two places. The first place, in terms of efficiency, is on GitHub when looking at the PR that needs reviewing. Using these labels you can immediately see what the purpose of the PR, and branch, is. Secondly, by having the naming convention, the branch is easily found and pulled on the reviewers machine for the local review.

I do however see that labels within the name might give a better git log overview in terms of what type of branches were merged and how that can be utilized when creating a changelog.

Reply:
> [...] not fond of the usage of labels within the branch name, `"refactor/SUM-XYZ-something-foo-bar"`, due to the extra step required when using the autocomplete feature of my terminal.

If you start with the ticket number you don't have this issue right? Primarily organising by ticket seems like it would make Jira easier too, and makes it easier to apply fixes / refactors / whatever related to a ticket name in a separate branch if needed, e.g.

- `FOO-123/feature/add-bar-factory`
- `FOO-123/fix/solve-duplication-bug-in-baz-strategy`
- `FOO-123/chore/upgrade-testing-framework-dependencies`